sympy/core/tests/test_exprtools.py

```diff
@@ -288,6 +288,10 @@ def test_factor_terms():
     assert factor_terms(e, sign=False) == e
     assert factor_terms(exp(-4*x - 2) - x) == -x + exp(Mul(-2, 2*x + 1, evaluate=False))
 
+    # sum tests
+    assert factor_terms(Sum(x, (y, 1, 10))) == x * Sum(1, (y, 1, 10))
```

Comment:
What happens to `factor_terms(Sum(x*y + x*y**2, (y, 1, 10)))`? Does it become `x*Sum(y*(y + 1), (y, 1, 10))`?